test/token/behaviors/CodexTitle.behavior.js

```diff
@@ -129,46 +129,212 @@ export default function shouldBehaveLikeCodexTitle(accounts) {
+
+          const { logs } = await this.token.modifyMetadataHashes(
+            0,
+            hashedMetadata.name,
+            '',
+            [hashedMetadata.imageBytes],
+            providerId,
+            providerMetadataId,
+          )
+
+          const tokenData = await this.token.getTokenById(0)
+
+          tokenData[0].should.be.equal(hashedMetadata.name)
+          tokenData[1].should.be.equal('0x0000000000000000000000000000000000000000000000000000000000000000')
+          tokenData[2].should.deep.equal([hashedMetadata.imageBytes])
+
+          // a Modified event is emitted when provider details are specified
+          logs.length.should.be.equal(1)
+
+          logs[0].event.should.be.eq('Modified')
+          logs[0].args._from.should.be.equal(creator)
+          logs[0].args._tokenId.should.be.bignumber.equal(0)
+          logs[0].args._newNameHash.should.be.equal(tokenData[0])
+          logs[0].args._newDescriptionHash.should.be.equal(tokenData[1])
+          logs[0].args._newImageHashes.should.deep.equal(tokenData[2])
+          logs[0].args._providerId.should.be.equal(providerId)
+          logs[0].args._providerMetadataId.should.be.equal(providerMetadataId)
+
         })
-      })
-    })
 
-    // TODO: Abstract this out into a generic pattern so it can also be used for description
-    describe('modifyNameHash', function () {
-      const newNameHash = web3.sha3('New name')
+        it('should update image hashes only', async function () {
+
+          const { logs } = await this.token.modifyMetadataHashes(
+            0,
+            '',
+            hashedMetadata.description,
+            newImageHashes,
+            providerId,
+            providerMetadataId,
+          )
+
+          const tokenData = await this.token.getTokenById(0)
+
+          tokenData[0].should.be.equal(hashedMetadata.name)
+          tokenData[1].should.be.equal(hashedMetadata.description)
+          tokenData[2].should.deep.equal(newImageHashes)
+
+          // a Modified event is emitted when provider details are specified
+          logs.length.should.be.equal(1)
+
+          logs[0].event.should.be.eq('Modified')
+          logs[0].args._from.should.be.equal(creator)
+          logs[0].args._tokenId.should.be.bignumber.equal(0)
+          logs[0].args._newNameHash.should.be.equal(tokenData[0])
+          logs[0].args._newDescriptionHash.should.be.equal(tokenData[1])
+          logs[0].args._newImageHashes.should.deep.equal(tokenData[2])
+          logs[0].args._providerId.should.be.equal(providerId)
+          logs[0].args._providerMetadataId.should.be.equal(providerMetadataId)
 
-      describe('when called by the owner', function () {
-        beforeEach(async function () {
-          await this.token.modifyNameHash(0, newNameHash)
         })
 
-        it('should succeed', async function () {
+        it('should update all hashes', async function () {
+
+          const newNameHash2 = web3.sha3('New name 2')
```

Comment:
Why are the second round of variables needed? Can't the outer ones be used instead?

Reply:
You're right, they're not. This was left over from the first pass at writing tests I did, where I thought state was saved between tests and I needed these to be different from the first set of variables. I'm still not really sure why the data updated data doesn't persist across tests 🤔